### PR TITLE
Add overload method to fetch LocationServiceHelper

### PR DESCRIPTION
**Pull request by syedowaisali90**
Hi, @vincent-karuri I have created a PR for review.

### CODE REVIEW

File: opensrp-app/src/main/java/org/smartregister/sync/helper/LocationServiceHelper.java [hunk 1]
```diff
@@ -405,6 +411,9 @@ public void fetchAllLocations() {
             JSONObject request = new JSONObject();
             request.put(IS_JURISDICTION, true);
             request.put(AllConstants.SERVER_VERSION, 0);
+            if (parentId != null) {
```
Comment (vincent-karuri) [inline opensrp-app/src/main/java/org/smartregister/sync/helper/LocationServiceHelper.java:414]: Could we use `StringUtils.isNotBlank(parentId)` instead?

File: opensrp-app/src/main/java/org/smartregister/sync/helper/LocationServiceHelper.java [hunk 2]
```diff
@@ -394,6 +395,11 @@ public String getFormattedBaseUrl() {
     }
 
     public void fetchAllLocations() {
+        fetchAllLocation(null);
+    }
+
+    public void fetchAllLocation(@Nullable String parentId) {
```
Comment (vincent-karuri) [inline opensrp-app/src/main/java/org/smartregister/sync/helper/LocationServiceHelper.java:401]: Please fix the typo here `fetchAllLocation` to `fetchAllLocations`.

**Comment by vincent-karuri** [comment]
@syedowaisali90 Were you able to test this change on the RDT app? You can publish the library locally and use it to test the changes: `./gradlew clean uploadArchives -PmavenLocal=true`. Make sure to change the `VERSION_NAME` in the `gradle.properties` to a higher value than is on the app.

File: opensrp-app/src/main/java/org/smartregister/sync/helper/LocationServiceHelper.java (hunk 1 above)
Comment (syedowaisali90) [reply opensrp-app/src/main/java/org/smartregister/sync/helper/LocationServiceHelper.java:414]: fixed.

**Comment by vincent-karuri** [comment]
The app crashes when launching the forms. Please verify that this feature works.

**Review by qiarie — CHANGES_REQUESTED** [review]
@syedowaisali90 
Add tests to cover the newly added functionality and update to latest master

File: opensrp-app/src/test/java/org/smartregister/sync/helper/LocationServiceHelperTest.java [hunk 3]
```diff
@@ -274,4 +290,70 @@ public void testFetchAllLocations() {
...
+
+        Mockito.doReturn(new Response<>(ResponseStatus.success, hierarchy))
+                .when(httpAgent).fetch(stringArgumentCaptor.capture());
+
+        LocationTree tree = locationServiceHelper.getLocationHierarchy(parentLocation);
+
+        String expectedUrl = MessageFormat.format("{0}{1}{2}", locationServiceHelper.getFormattedBaseUrl(), LocationServiceHelper.LOCATION_HIERARCHY_URL, parentLocation);
+        assertEquals(expectedUrl, stringArgumentCaptor.getAllValues().get(0));
```
Comment (vincent-karuri) [inline opensrp-app/src/test/java/org/smartregister/sync/helper/LocationServiceHelperTest.java:349]: You can use `stringArgumentCaptor.getValue()` here and in the other places.

File: opensrp-app/src/test/java/org/smartregister/sync/helper/LocationServiceHelperTest.java [hunk 4]
```diff
@@ -274,4 +290,70 @@ public void testFetchAllLocations() {
...
+        assertNull(tree);
+    }
+
+    @Test
+    public void testGetLocationHierarchyReturnsNullForInvalidParentLocationId() {
```
Comment (vincent-karuri) [inline opensrp-app/src/test/java/org/smartregister/sync/helper/LocationServiceHelperTest.java:324]: Doesn't seem like the method returns null for invalid parent location id? https://github.com/OpenSRP/opensrp-client-core/pull/663/files#diff-7d2c1e8994b4bf118233996655641ad90f6f71d2f8145fcea0e1bf67136a37e7R334

Comment (qiarie) [reply opensrp-app/src/test/java/org/smartregister/sync/helper/LocationServiceHelperTest.java:324]: Updated test method name to reflect the actual test

**Review by vincent-karuri — APPROVED** [review]

**Review by qiarie — APPROVED** [review]

**Pull request merged**